### PR TITLE
View Components

### PR DESCRIPTION
**Pull request by colby-swandale**
This PR is a starting point described in #1156  to break up the UI into a set of components that can be reused, tested and documented in design systems like Storybook.

This PR moves the header & logo into a new View Component as they are. The header will need to be further broken up into smaller components, and further PRs will be made to convert partials & any common elements into a component.

### CODE REVIEW

**Comment by natematykiewicz** [comment]
This gem seems very interesting for previewing all the components.

https://github.com/allmarkedup/lookbook

File: app/components/logo_component.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class LogoComponent < ViewComponent::Base
+
+end
```
Comment (AlexWayfer) [inline app/components/logo_component.rb:5]: RuboCop would not like it.

**Review by AlexWayfer — APPROVED** [review]
I see no critical problems.

**Comment by natematykiewicz** [comment]
Delete `test/components/stories/logo_component_stories.rb`?

File: app/components/header_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<header data-controller="header" class="flex items-center h-16 text-white bg-red-600 dark:bg-gray-700 <%= "fixed inset-x-0 z-50 top-0" if @fixed_nav %>">
+  <nav class="w-full max-w-screen-xl mx-auto px-3" role="navigation" aria-label="main navigation">
+    <div class="flex items-center justify-between space-x-4">
+      <div class="flex-none -mx-1">
```
[the remark this answers is not in the document]
Comment (colby-swandale) [reply app/components/header_component.html.erb:4]: It's a [negative margin](https://tailwindcss.com/docs/margin#using-negative-values)

[the remark this answers is not in the document]
Comment (natematykiewicz) [reply app/components/header_component.html.erb:4]: Today I Learned

**Review by natematykiewicz — APPROVED** [review]

**Comment by colby-swandale** [comment]
@natematykiewicz @AlexWayfer, thank you for your reviews 🙇🏻

**Pull request merged**